app/services/usps_in_person_proofing/date_validator.rb

```diff
@@ -0,0 +1,37 @@
+module UspsInPersonProofing
+  # Validator that can be attached to a form or other model
+  # to verify that specific supported fields are comparable to other dates.
+  # Since it's a subclass of ComparisonValidator, it share the same options.
+  # == Example
+  #
+  #   validates_with UspsInPersonProofing::DateValidator,
+  #     attributes: [:dob],
+  #     message: "error",
+  #     less_than_or_equal_to: ->(_rec) {
+  #         Time.zone.today - IdentityConfig.store.idv_min_age_years.years
+  #     }
+  #     ....
+  #
+  class DateValidator < ActiveModel::Validations::ComparisonValidator
+    private
+
+    def prepare_value_for_validation(value, _record, _attr_name)
```

Comment:
for my own understanding : 
1. why do we want to include `record` and `attr_name` as params here if we don't intend to use them?
2. why have this method (`prepare_value_for_validation`) that just calls another method (`val_to_date`)? Do we need both?